R/zchunk_LA101.en_bal_IEA.R

```diff
@@ -39,72 +39,184 @@ module_energy_LA101.en_bal_IEA_DISABLED <- function(command, ...) {
-    # There's also a `same_precursors_as(x)` you can use
-    # If no precursors (very rare) don't call `add_precursor` at all
-    tibble() %>%
-      add_title("descriptive title of data") %>%
-      add_units("units") %>%
-      add_comments("comments describing how data generated") %>%
-      add_comments("can be multiple lines") %>%
+    L100.IEA_en_bal_ctry_hist <- get_data(all_data, "temp-data-inject/L100.IEA_en_bal_ctry_hist")
+    ###############################################################################################################
+    # Data in long format.
+    L100.IEA_en_bal_ctry_hist %>%
+      gather(year, value, -iso, -FLOW, -PRODUCT) %>%
+      mutate(year=as.integer(substr(year,2,5))) -> L101.IEA_en_bal_ctry_hist
+
+    # Add IEA data to main tibble
+    L101.IEA_en_bal_ctry_hist %>%
+      left_join_error_no_match(select(iso_GCAM_regID, iso, GCAM_region_ID), by = "iso") %>%
+      # It is OK having NA at this stage since not every record has a match. NAs will be removed in the next step
+      left_join(select(IEA_flow_sector, FLOW=flow_code, sector), by ="FLOW") %>%
+      left_join(select(IEA_product_fuel, fuel, PRODUCT = product), by ="PRODUCT") %>%
+      left_join(select(IEA_flow_sector, FLOW = flow_code, conversion), by ="FLOW") %>%
+      na.omit() -> L101.IEA_en_bal_ctry_hist
+
+    # Rename biofuel fuel name for appropiate sectors
+    L101.IEA_en_bal_ctry_hist %>%
+      mutate( fuel = if_else(fuel == "biomass_tradbio" & sector != "in_bld_resid", "biomass", fuel)) %>%
+      left_join_error_no_match(select(A_regions, tradbio_region, GCAM_region_ID), by = "GCAM_region_ID") %>%
+      mutate(fuel = if_else(fuel == "biomass_tradbio" & tradbio_region == 0, "biomass", fuel)) %>%
+      select(- tradbio_region) -> L101.IEA_en_bal_ctry_hist
+
+    # In some countries, "gas works gas" is produced from coal. This is calibrated (coal gasification), so re-name the relevant sectors
+    # Rename the sector and the fuel: where the sector is gas works and the fuel is coal, this is the input to gas processing
+    L101.IEA_en_bal_ctry_hist %>%
+      mutate(sector= if_else(sector == "net_gas works" & fuel == "coal", "in_gas processing", sector)) %>%
+      #Where the sector is gas works and the fuel is not coal, this is industry/energy transformation
+      mutate(sector= if_else(sector == "net_gas works" & fuel != "coal", "net_industry_energy transformation", sector)) -> L101.IEA_en_bal_ctry_hist
+
+    # Create aux tibble for IEA_sector_fuel_modifications with renamed values for Re-setting sector-fuel combinations as specified in IEA_sector_fuel_modifications
+    IEA_sector_fuel_modifications %>%
+      mutate(sector = if_else(is.na(sector), "none", sector)) %>%
+      mutate(fuel = if_else(is.na(fuel), "none", fuel)) %>%
+      rename(sector_IEA = sector, fuel_IEA = fuel) %>%
+      select(sector_IEA, fuel_IEA, sector = sector_initial, fuel = fuel_initial, conversion_IEA = conversion) %>%
```

Comment:
Actually, just compute (in `mutate`) the appropriate names. Why rename subsequently?